src/Microsoft.ML.Transforms/FourierDistributionSampler.cs

```diff
@@ -5,205 +5,252 @@
 using System;
 using Microsoft.ML;
 using Microsoft.ML.CommandLine;
-using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Data;
 using Microsoft.ML.Internal.Utilities;
 using Microsoft.ML.Model;
+using Microsoft.ML.Numeric;
 using Microsoft.ML.Transforms;
 
-[assembly: LoadableClass(typeof(GaussianFourierSampler), typeof(GaussianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Gaussian Kernel", GaussianFourierSampler.LoadName, "Gaussian")]
+[assembly: LoadableClass(typeof(GaussianRngGenerator), typeof(GaussianRngGenerator.Options), typeof(SignatureRngGenerator),
+    "Gaussian Kernel", GaussianRngGenerator.LoadName, "Gaussian")]
 
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), typeof(LaplacianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Laplacian Kernel", LaplacianFourierSampler.RegistrationName, "Laplacian")]
+[assembly: LoadableClass(typeof(LaplacianRngGenerator), typeof(LaplacianRngGenerator.Options), typeof(SignatureRngGenerator),
+    "Laplacian Kernel", LaplacianRngGenerator.LoadName, "Laplacian")]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(GaussianFourierSampler), null, typeof(SignatureLoadModel),
-    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(GaussianRngGenerator.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianRngGenerator.RandomNumberGenerator.LoaderSignature)]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), null, typeof(SignatureLoadModel),
-    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(LaplacianRngGenerator.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianRngGenerator.RandomNumberGenerator.LoaderSignature)]
 
 // REVIEW: Roll all of this in with the RffTransform.
 namespace Microsoft.ML.Transforms
 {
     /// <summary>
-    /// Signature for an IFourierDistributionSampler constructor.
+    /// Signature for a <see cref="RngGeneratorBase"/> constructor.
     /// </summary>
     [BestFriend]
-    internal delegate void SignatureFourierDistributionSampler(float avgDist);
+    internal delegate void SignatureRngGenerator();
 
-    public interface IFourierDistributionSampler : ICanSaveModel
+    public abstract class RngGeneratorBase
```

Comment:
In general, .NET APIs prefer to write out the names. `Rng`. Also I assume the `g` in `Rng` is "generator", so it is a little redundant.

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions#using-abbreviations-and-acronyms](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions#using-abbreviations-and-acronyms)

X DO NOT use abbreviations or contractions as part of identifier names.

For example, use GetWindow rather than GetWin.

___________

Can we add some public XML doc to this new abstract interface?

Reply:
Maybe it would be good to tie this with `Fourier` as well? Since this is in the `Microsoft.ML.Transforms` namespace, having a `RngGeneratorBase` class seems a bit broad, especially one that doesn't have any public members on it.

---
In reply to: [259486033](https://github.com/dotnet/machinelearning/pull/2698#discussion_r259486033) [](ancestors = 259486033)

Reply:
Yes, I was going to mention that I'm not happy with these class names... I wrote an explanation about these classes in a separate comment, I'll also add some XML comments and post another iteration for this PR.


---
In reply to: [259486033](https://github.com/dotnet/machinelearning/pull/2698#discussion_r259486033) [](ancestors = 259486033)